app/gui/view/graph-editor/src/component/visualization/container.rs

```diff
@@ -125,29 +134,63 @@ pub mod background {
 // === Frp ===
 // ===========
 
-ensogl::define_endpoints! {
+/// Indicates the visibility state of the visualisation.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum ViewState {
+    /// Visualisation is permanently enabled and visible in thw graph editor. It is attached to a
+    /// single node and can be moved and interacted with when selected.
+    Enabled,
+    /// Visualisation is disabled and hidden in the graph editor.
+    Disabled,
+    /// Visualisation is temporarily enabled and visible in the graph editor. It should be placed
```

Comment:
Is any difference between "enabled" and "visible"? Before, "enabled" meant that the visualization "eye" was toggled on. Visualization could be visible and not enabled (in case of Preview) and enabled but not visible (if there's error on the node - I think we miss that variant here).

Reply:
This logic is now decoupled; thus, the visualization is either visible or not. While the button's state is managed independently. I will double-check the button has the correct state during preview.